### PR TITLE
add definition of a VO

### PR DESCRIPTION
**Pull request by paulmillar**
Motivation:

The document makes many references to VOs but neither defining for what "VO" stands, nor describing what is a virtual organisation.

This is bad because the document makes a destinction between VOs (as top-level collection of users) and group (as non-top-level collection of useres), so the definition matters.

Modification:

Modify first paragraph to use "virtual organisation" instead of VO.

Add definition of a VO.

Result:

It becomes easier to understand whether a group of people constitute a VO.

Closes: #38

### CODE REVIEW

File: profile.md [hunk 1]
```diff
@@ -37,7 +37,16 @@ _Authored by the WLCG AuthZ Working Group_
 
 # Introduction
 
-This document describes how WLCG users may use the available geographically distributed resources without X.509 credentials.  In this model, clients are issued with bearer tokens; these tokens are subsequently used to interact with resources.  The tokens may contain authorization groups and/or capabilities, according to the preference of the VO, applications and relying parties. 
+This document describes how WLCG users may use the available geographically distributed resources without X.509 credentials.  In this model, clients are issued with bearer tokens; these tokens are subsequently used to interact with resources.  The tokens may contain authorization groups and/or capabilities, according to the preference of the virtual organisation, applications and relying parties.
+
+A virtual organisation (VO) is a group of users that are
+self-managing, often with internal structure (such as groups and
```
Comment (DrDaveD) [inline profile.md:43]:
```suggestion
independently managed, often with internal structure (such as groups and
```
At Fermilab we have a group that provides services to many VOs (both those that we consider sub-VOs and full VOs), including the management of computing infrastructures.  So `self-managed` isn't accurate for them.

File: profile.md [hunk 2]
```diff
@@ -37,7 +37,16 @@ _Authored by the WLCG AuthZ Working Group_
...
+roles).  The members of a VO work towards a common goal; for example,
+maximising the scientific output from a specific scientific
+instrument.  A VO has well understood membership criteria, which the
+members of the VO manage themselves.  Their work is often supported
+through the use of resources located multiple facilities.  Within
```
Comment (DrDaveD) [inline profile.md:48]:
```suggestion
through the use of resources located at multiple facilities.  Within
```

File: profile.md [hunk 3]
```diff
@@ -37,7 +37,16 @@ _Authored by the WLCG AuthZ Working Group_
...
 # Introduction
 
-This document describes how WLCG users may use the available geographically distributed resources without X.509 credentials.  In this model, clients are issued with bearer tokens; these tokens are subsequently used to interact with resources.  The tokens may contain authorization groups and/or capabilities, according to the preference of the VO, applications and relying parties. 
+This document describes how WLCG users may use the available geographically distributed resources without X.509 credentials.  In this model, clients are issued with bearer tokens; these tokens are subsequently used to interact with resources.  The tokens may contain authorization groups and/or capabilities, according to the preference of the virtual organisation, applications and relying parties.
+
+A virtual organisation (VO) is a group of users that are
+self-managing, often with internal structure (such as groups and
+roles).  The members of a VO work towards a common goal; for example,
+maximising the scientific output from a specific scientific
+instrument.  A VO has well understood membership criteria, which the
```
Comment (DrDaveD) [inline profile.md:46]:
```suggestion
instrument  or providing shared services to multiple related projects.
A VO has well understood membership criteria, which the
```

**Review by DrDaveD — CHANGES_REQUESTED** [review]

**Comment by msalle** [comment]
I think the only thing we really need is a administrative body that has control over the scope/namespace of the attributes used in a certain token. Perhaps we can formulate the definition of VO around that, and make it much more abstract.
I feel we otherwise use a lot of specifics and unnecessary baggage from the VOMS era while a more abstract definition is probably sufficient.

File: profile.md (hunk 3 above)
Comment (paulmillar) [reply profile.md:46]: Thanks for the suggestion.  I've updated the text.

I was wondering if you have a good example of a service-providing VO, that I include in the list of example VOs?

File: profile.md (hunk 1 above)
Comment (paulmillar) [reply profile.md:43]: Thanks.

File: profile.md (hunk 3 above)
Comment (DrDaveD) [reply profile.md:46]: `fermilab` and `osg` are the two that I know about.

Comment (paulmillar) [reply profile.md]: Thanks

**Comment by paulmillar** [comment]
Hi @msalle , I think you're right about "administrative body that has control over the scope/namespace of the attributes".

I think this is related to the idea of VOs having their own, independent identity.  In the current definition, there are a few places that hint at this (e.g., the phrase "a group of users that are independently managed" and "A VO has well understood membership criteria, which the members of the VO manage themselves").  Perhaps we could rephrase this to make it clearer.

I'm a little nervous about defining VOs in terms of what they can do (under current technology) because 1) technology changes, and 2) I think it risks putting the cart before the horse: use-cases (along with corresponding definitions) should drive technology, rather than the other way around.

Perhaps we could phrase your "VOs have authority on asserted attributes" (if I've captured that correctly) more as a consequence of our definition of a VO?

**Comment by msalle** [comment]
> Hi @msalle , I think you're right about "administrative body that has control over the scope/namespace of the attributes".
> 
> I think this is related to the idea of VOs having their own, independent identity. In the current definition, there are a few places that hint at this (e.g., the phrase "a group of users that are independently managed" and "A VO has well understood membership criteria, which the members of the VO manage themselves"). Perhaps we could rephrase this to make it clearer.
> 
> I'm a little nervous about defining VOs in terms of what they can do (under current technology) because 1) technology changes, and 2) I think it risks putting the cart before the horse: use-cases (along with corresponding definitions) should drive technology, rather than the other way around.
> 
> Perhaps we could phrase your "VOs have authority on asserted attributes" (if I've captured that correctly) more as a consequence of our definition of a VO?

I think it's kind of the reverse: I think we shouldn't try to define a VO since there are many different interpretations/meanings/definitions going around and ultimately we don't need the concept in the document (also we should not try to/don't need to reformulate the X509/RFC3820/VOMS model in terms of tokens, so we don't per se need all the old concepts).
What we need is the administrative body having the authority on the asserted attributes. So maybe we can just remove the whole term VO from the document?

**Comment by paulmillar** [comment]
So we would update the document by adding something like

> An attribute authority is a group of people who are authoritative in asserting group membership and authorisation statements.  An attribute authority may have policies that restrict  who can make such assertions. In several scientific communities, the attribute authority is called a virtual organisation (VO).

We would also replace all mentions of "VO" with "attribute authority".

**Comment by maarten-litmaath** [comment]
Hi all,
I have just submitted #47 with an alternative VO definition and description, for several reasons:

* I think a VO is more than just an attribute authority, while, as far as tokens are concerned, it **acts** as such;
* I would like us to keep being able to use the familiar notion of a "VO" throughout the document;
* I have various concerns about the previously proposed texts, which I have tried to address accordingly.

What do you think?

**Comment by maarten-litmaath** [comment]
Superseded by #47 .
